tests/components/technove/test_config_flow.py

```diff
@@ -102,3 +104,160 @@ async def test_full_user_flow_with_error(
+
+
+async def test_zeroconf_connection_error(
+    hass: HomeAssistant, mock_technove: MagicMock
+) -> None:
+    """Test we abort zeroconf flow on TechnoVE connection error."""
+    mock_technove.update.side_effect = TechnoVEConnectionError
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": SOURCE_ZEROCONF},
+        data=zeroconf.ZeroconfServiceInfo(
+            ip_address=ip_address("192.168.1.123"),
+            ip_addresses=[ip_address("192.168.1.123")],
+            hostname="example.local.",
+            name="mock_name",
+            port=None,
+            properties={CONF_MAC: "AA:AA:AA:AA:AA:BB"},
+            type="mock_type",
+        ),
+    )
+
+    assert result.get("type") == FlowResultType.ABORT
+    assert result.get("reason") == "cannot_connect"
+
+
+@pytest.mark.usefixtures("mock_technove")
+async def test_user_station_exists_abort(
+    hass: HomeAssistant, mock_config_entry: MockConfigEntry
+) -> None:
+    """Test we abort zeroconf flow if TechnoVE station already configured."""
+    mock_config_entry.add_to_hass(hass)
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": SOURCE_USER},
+        data={CONF_HOST: "192.168.1.123"},
+    )
+
+    assert result.get("type") == FlowResultType.ABORT
+    assert result.get("reason") == "already_configured"
+
+
+@pytest.mark.usefixtures("mock_technove")
+async def test_zeroconf_without_mac_station_exists_abort(
+    hass: HomeAssistant,
+    mock_config_entry: MockConfigEntry,
+) -> None:
+    """Test we abort zeroconf flow if TechnoVE station already configured."""
+    mock_config_entry.add_to_hass(hass)
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": SOURCE_ZEROCONF},
+        data=zeroconf.ZeroconfServiceInfo(
+            ip_address=ip_address("192.168.1.123"),
+            ip_addresses=[ip_address("192.168.1.123")],
+            hostname="example.local.",
+            name="mock_name",
+            port=None,
+            properties={},
+            type="mock_type",
+        ),
+    )
+
+    assert result.get("type") == FlowResultType.ABORT
+    assert result.get("reason") == "already_configured"
+
+
+async def test_zeroconf_with_mac_station_exists_abort(
+    hass: HomeAssistant, mock_config_entry: MockConfigEntry
+) -> None:
+    """Test we abort zeroconf flow if TechnoVE station already configured."""
+    mock_config_entry.add_to_hass(hass)
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": SOURCE_ZEROCONF},
+        data=zeroconf.ZeroconfServiceInfo(
+            ip_address=ip_address("192.168.1.123"),
+            ip_addresses=[ip_address("192.168.1.123")],
+            hostname="example.local.",
+            name="mock_name",
+            port=None,
+            properties={CONF_MAC: "AA:AA:AA:AA:AA:BB"},
+            type="mock_type",
+        ),
+    )
+
```

Comment:
Should this also verify that the quick path is taken, i.e., no `_async_get_station` gets called?

Reply:
Done. Does this look good? I tested it by commenting out the `_abort_if_unique_id_configured` from the quick path in config_flow and it seems to be working well for this purpose.

Reply:
Sounds good to me :+1: